Training_Center_Task_10/Tests.cs

```diff
@@ -0,0 +1,301 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.DevTools.V104.Debugger;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.Support.UI;
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+
+namespace Training_Center_Task_10
+{
+    [TestClass]
+    public class Tests
+    {
+        public IWebDriver _driver;
+        [TestInitialize]
+        public void Init()
+        {
+            this._driver = new FirefoxDriver();
+
+            _driver.Manage().Window.Maximize();
+        }
+
+        [TestCleanup]
+        public void DriverQuit()
+        {
+            _driver.Quit();
+        }
+
+        [DataTestMethod]
+        //[DataRow("r0b3rt.sm1th", "R0b3rt!@", "r0b3rt.sm1th")]
+        //[DataRow("test.tester.10", "testertest10", "Test")]
+        [DataRow("bignevx", "Parolata007!", "bignevx")]
+        [DataRow("bignevx", "Parolata007!", "bignevx")]
+        [Description("Tasks 1 - 4")]
+
+        [TestMethod]
+        public void Login(string username, string password, string name)
+        {
+            _driver.Navigate().GoToUrl("https://mail.yandex.com/");
+
+            //Click Accept button
+            IWebElement acceptButton = _driver.FindElement(By.CssSelector("button[data-id='button-all']"));
+            acceptButton.Click();
+
+            //Click LogIn button
+            IWebElement loginButton = _driver.FindElement(By.CssSelector
+               ("button[class='Button2 Button2_size_m Button2_view_action Button2_weight_500 Button_Vd8eu21iIVyRdyjGPVfYF PSHeader-NoLoginButton']"));
+            loginButton.Click();
+            
+            //Fill in username
+            IWebElement usernameTextbox = _driver.FindElement(By.Id("passp-field-login"), 10);
+            usernameTextbox.SendKeys(username);
+
+            //Click LogIn button
+            IWebElement loginButtonUsername = _driver.FindElement(By.CssSelector("button[id='passp:sign-in']"), 10);
+            loginButtonUsername.Click();
+
+            //Fill in password
+            IWebElement passwordTextbox = _driver.FindElement(By.Id("passp-field-passwd"), 10);
+            passwordTextbox.Click();
+            passwordTextbox.SendKeys(password);
+
+            //Click LogIn button
+            IWebElement loginButtonPassword = _driver.FindElement(By.CssSelector("button[id='passp:sign-in']"), 10);
+            loginButtonPassword.Click();
+
+            _driver.Wait(5000);
```

Comment:
using thread.sleep is one of bad practices. please  try another way to wait for something